src/Raven.Server/Documents/Replication/Incoming/IncomingReplicationHandler.cs

```diff
@@ -743,11 +743,23 @@ protected virtual void SaveSourceEtag(DocumentsOperationContext context)
 
             private void RecordDatabaseChangeVector(DocumentsOperationContext context)
             {
+                IAbstractIncomingReplicationHandler incomingHandler = null;
                 try
                 {
-                    var stats = context.DocumentDatabase.ReplicationLoader.IncomingHandlers.
-                        Single(i => i.ConnectionInfo.SourceDatabaseId == _replicationInfo.SourceDatabaseId)
-                        .GetLatestReplicationPerformance();
+                    try
+                    {
+                        incomingHandler = context.DocumentDatabase.ReplicationLoader.IncomingHandlers.Single(i =>
```

Comment:
cant we do just `if(context.DocumentDatabase.ReplicationLoader.IncomingHandlers.Count ==0) log & return;`
why do we want to do try-catch here ?

Reply:
I think we should keep the try-catch here because it covers two cases:
1. `IncomingHandlers` is empty
2. `Single()` throws an exception because we have more than one match 

the last one is super rare, but maybe to stay safe let's keep it?

Reply:
I'd prefer to handle the first case explicitly as @garayx suggested. Regarding second case, is it a valid case (even if rare)? If they then let's handle it explicitly as well.

Reply:
Then we can get rid of generic `try / catch` at all

Reply:
done